src/ConstructiveSolidGeometry/SurfacePrimitives/Polygon.jl

```diff
@@ -55,7 +55,8 @@ function sample(p::Polygon{N,T}, spacing::T)::Vector{CartesianPoint{T}} where {N
 end
 
 function extremum(p::Polygon{N,T})::T where {N,T}
-    c = sum(p.points)/N
+    c = cartesian_zero + mean(reinterpret(CartesianVector{T}, p.points))  
+    # c = mean(p.points)
```

Comment:
I can't use `c = mean(p.points)` here, because `p.points` is a `StaticVector{CartesianPoint{T}}` and `mean` here then defaults to the [`mean` method defined in `StaticArraysStatisticsExt` ](https://github.com/JuliaArrays/StaticArrays.jl/blob/a9333fa830d27b85d6c8aa284d32df1b6de0634f/ext/StaticArraysStatisticsExt.jl#L15).

@oschulz any way we can "force" SSD/CSG to use the method defined for `CartesianPoint{T}` that avoids using `+`?

To reproduce the error:
```julia
using SolidStateDetectors, StaticArrays, Statistics; T = Float32
points = @SVector ([CartesianPoint{T}(0,0,0), CartesianPoint{T}(1,1,1)])
mean(points) # throws an error
```

Reply:
Or do you mean using `AbstractVector` here instead of `AbstractArray`?

Reply:
It should actually work for any collection, doesn't have to be a vector. mean/barycenter doesn't have any connection to indexing scheme, after all. Even sets are fine. Maybe it's easiest if we really use a separate operation:

```
function barycenter(points)
    vectors = points .- Ref(cartesian_zero)
    mean_vector = mean(vectors)
    R = eltype(points)
    return convert(R, mean_vector + cartesian_zero)::R
end
```

Reply:
Let's reduce memory allocation:

```julia
barycenter(X) = _barycenter_impl(eltype(xs), X)

function _barycenter_impl(::Type{T}, points) where {T<:AbstractCoordinatePoint}
    mean_vector = mean(Base.Fix2(-, cartesian_zero), points)
    return convert(T, mean_vector + cartesian_zero)::T
end
```

Reply:
The problem is that many packages specialize `mean` by container type, so we really can't specialize it by element-type for a large range of container types. So it's better to use a more specific method name, I think, esp. since it's the one used on maths for this kind of thing.